robottelo/hosts.py

```diff
@@ -1674,14 +1675,31 @@ def get_caller():
                 if frame.function.startswith('test_'):
                     return frame.function
 
-        return Session(
-            session_name=testname or get_caller(),
-            user=user or settings.server.admin_username,
-            password=password or settings.server.admin_password,
+        test_name = testname or get_caller()
+
+        if getattr(self, 'organization') and getattr(self, 'location') and user is None:
+            login = f'{test_name}_{gen_string("alphanumeric")}'
+            password = gen_string('alphanumeric')
+            new_user = self.api.User(
+                admin=True,
+                default_organization=self.organization,
+                default_location=self.location,
+                description=f'created automatically by airgun for module "{test_name}"',
+                login=login,
+                password=password,
+            ).create()
+            new_user.password = password
+
+        yield Session(
+            session_name=test_name,
+            user=user or new_user.login,
+            password=password or new_user.password,
             url=url,
             hostname=self.hostname,
             login=login,
         )
+        if user is None:
+            new_user.delete()
```

Comment:
creating a user and deleting that user feels cumbersome as this will run every time for each test, we have to handle this a bit better way. Maybe we can use module_user injecting as part of the host object. So that we do no need to delete that user.

Reply:
yes, I felt the same but user create/delete will be very quick with API, and even when using module_user fixtures for each module will be same thing, instead we can have some session scoped airgun_automation user created so it'll be per xdist worker, so for each test every time we need to update its organization/location, and user creation/deletion  will be once per session, wdyt?

Reply:
yes, I felt the same but user create/delete will be very quick with API which still invalidates the point I'm making here. We should not want to create the user every time. I ack on the module user to create a fixture, that way we will be sharing that user across multiple tests with tests. I do also ack on updating the default_org and default location as part of injecting the ui_session user. (we even can use the admin user here in that case as well, if user parameter is None)